valueset.go

```diff
@@ -326,12 +328,19 @@ func Execute(ctx context.Context, tx neo4j.ExplicitTransaction, cfg Config, oldN
 	if len(inserts) > 0 {
 		for _, insert := range groupByLabel(inserts) {
 			unwindData := map[string]interface{}{"nodes": mapNodesetData(insert.data)}
-			query := fmt.Sprintf("UNWIND $nodes AS node MERGE (n%s {`%s`: node.ID}) SET n=node.Properties WITH n MATCH(root:`NODESET` {`%s`: %s}) MERGE(root)-[:%s]->(n) MERGE(n)-[:%s]->(root)",
-				insert.labelExpr, cfg.Shorten(ls.EntityIDTerm), cfg.Shorten(ls.EntityIDTerm), quoteStringLiteral(newNodeset.ID), newNodeset.ID, newNodeset.ID)
+			query := fmt.Sprintf("UNWIND $nodes AS node CREATE (n%s {`%s`: node.ID}) SET n=node.Properties",
+				insert.labelExpr, cfg.Shorten(ls.EntityIDTerm))
 
 			if _, err := tx.Run(ctx, query, unwindData); err != nil {
 				return err
 			}
+			// create edges
+			unwindData = map[string]interface{}{"nodes": mapNodesetData(insert.data)}
```

Comment:
Edges should be inserted based on the newNodeset, and unconditionally (not just for inserted nodes).
Edges should be create with MERGE, not CREATE, or this will add duplicate edges.
You'll need to group all nodes in NewNodeset by label, and add edges using group of labels and entityId